### PR TITLE
doc: give hint how to rebase on master

### PR DESCRIPTION
**Pull request by bachp**
###### Motivation for this change

Give people a hint how to easily rebase their changes on master
###### Things done
- [x] Tested using nix-build doc

---

### CODE REVIEW

**Comment by ericsagnes** [comment]
That rebase command feels overcomplicated.
Isn't `git rebase master` enough to rebase on `master`?

**Comment by bachp** [comment]
The problem is that if you are on a release branch and you do just `git rebase master` you will also rebase all the commits that were in that release branch but not on master.

**Comment by ericsagnes** [comment]
> The problem is that if you are on a release branch and you do just git rebase master you will also rebase all the commits that were in that release branch but not on master.

Indeed, that make sense with the previous part of the documentation.

A few nitpicks:
- Where 0998212 is the commit of your current installation -> Where 0998212 is the upstream branch name or commit (As current installation commit and branch upstream commit / branch name might be different.)
- in your branche. -> in your branch.
- it might be worth to say that if the upstream branch is master, `git rebase master` is enough.

File: doc/submitting-changes.xml [hunk 1]
```diff
@@ -178,9 +178,18 @@ Additional information.
 </listitem>
 
 <listitem>
-<para>Rebase you branch against current <command>master</command>.</para>
+<para>Rebase you branch against current <command>master</command>.
```
Comment (rasendubi) [inline doc/submitting-changes.xml]: s/you/your

**Comment by FRidh** [comment]
So write first `git rebase master` but note that this only works when you've branched from master. If, however, you are on a release, then because .... that won't work and you need <the more complicated rebase>. That way, one that isn't too familiar with `git rebase` has a better idea what he or she is doing and why.

**Comment by bachp** [comment]
I think if you follow the instructions which state that you should not work on master it is the more common case that you need a rebase onto as a simple rebase would most probably add some additional commits.

**Comment by jagajaga** [comment]
Maybe let's copy git tutorial to our guides?
This thing is really nix unrelated.

**Pull request closed without merging**